### ISSUE TITLE
Scoring functions

### ISSUE THREAD

**Issue by bellet** [metric-learn/metric-learn#165]
Besides the default `score` method of our estimators, we should properly test and document the fact that one can use [various metrics](https://scikit-learn.org/stable/modules/classes.html#module-sklearn.metrics) to evaluate our supervised and weakly-supervised metric learners when using sklearn's tools such as `model_selection.GridSearchCV` and `model_selection.cross_val_score` through their `scoring` parameter.

This should be done after dealing with #131, which will introduce a predict method for weakly supervised pairwise metric learners.
